src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,601 @@
+        iv::InterfaceValues,
+        cell_here::AbstractCell, coords_here::AbstractVector{Vec{dim, T}}, face_here::Int,
+        cell_there::AbstractCell, coords_there::AbstractVector{Vec{dim, T}}, face_there::Int
+    ) where {dim, T}
+
+    # reinit! the here side as normal
+    reinit!(iv.here, coords_here, face_here)
+    dim == 1 && return reinit!(iv.there, coords_there, face_there)
+    # Transform the quadrature points from the here side to the there side
+    iv.there.current_face[] = face_there
+    interface_transformation = InterfaceTransformation(cell_here, cell_there, face_here, face_there)
+    quad_points_a = getpoints(iv.here.qr, face_here)
+    quad_points_b = getpoints(iv.there.qr, face_there)
+    transform_interface_points!(quad_points_b, quad_points_a, interface_transformation)
+    @boundscheck checkface(iv.there, face_there)
+    # TODO: This is the bottleneck, cache it?
+    @assert length(quad_points_a) <= length(quad_points_b)
+    # Re-evalutate shape functions in the transformed quadrature points
+    for qp in 1:length(quad_points_a), i in 1:getnbasefunctions(iv.there)
+        iv.there.dNdξ[i, qp, face_there], iv.there.N[i, qp, face_there] = shape_gradient_and_value(iv.there.func_interp, quad_points_b[qp], i)
+    end
+    for qp in 1:length(quad_points_a), i in 1:getngeobasefunctions(iv.there)
+        iv.there.dMdξ[i, qp, face_there], iv.there.M[i, qp, face_there] = shape_gradient_and_value(iv.there.geo_interp, quad_points_b[qp], i)
+    end
+    # reinit! the "there" side
+    reinit!(iv.there, coords_there, face_there)
+    return iv
+end
+
+"""
+    getnormal(iv::InterfaceValues, qp::Int; here::Bool=true)
+
+Return the normal vector in the quadrature point `qp` on the interface. If `here = true`
+(default) the outward normal to the "here" element is returned, otherwise the outward normal
+to the "there" element.
+"""
+function getnormal(iv::InterfaceValues, qp::Int; here::Bool=true)
+    # TODO: Does it make sense to allow the kwarg here? You can juse negate the vector
+    #       yourself since getnormal(iv, qp; here=false) == -getnormal(iv, qp; here=true).
+    return here ? iv.here.normals[qp] : iv.there.normals[qp]
+end
+
+"""
+    function_value(iv::InterfaceValues, q_point::Int, u_here::AbstractVector, u_there::AbstractVector; here::Bool)
+    function_value(iv::InterfaceValues, q_point::Int, u_here::AbstractVector, range_here, u_there::AbstractVector, range_there; here::Bool)
+
+Compute the value of the function in quadrature point `q_point` on the "here" (`here=true`)
+or "there" (`here=false`) side of the interface. `u_here` and `u_there` are the values of
+the degrees of freedom for the respeciv element.
+
+
+TODO: Complete the docs here.
+is a vector with values
+for the degrees of freedom. For a scalar valued function, `u` contains scalars.
+For a vector valued function, `u` can be a vector of scalars (for use of `VectorValues`)
+or `u` can be a vector of `Vec`s (for use with ScalarValues).
```

Comment:
This part is outdated, right? IIUC vector valued problems use `Vec` not scalars only and `(Vector/Scalar)Values)` are deprecated since #708

Reply:
I think it is still possible to allow `Vector{Vec}`, but it was suggested to be removed in https://github.com/Ferrite-FEM/Ferrite.jl/pull/698 (but didn't happen, should it @fredrikekre ?). The standard way, however, is that `u` is a `Vector` of scalars for both scalar and vector function interpolations.

Reply:
Thanks for the clarification!
I thought it was similar to `shapeValues` such that `u` is a vector of  `Vec` if we use vectorized interpolation.

Reply:
I think we should remove it, and therefore not add it to InterfaceValues.